### PR TITLE
Xpetra: Making previous assumptions about Xpetra configuration very clear

### PR DESCRIPTION
**Pull request by csiefer2**
Auto-PR for SHA ad8ea7e

### CODE REVIEW

File: packages/xpetra/CMakeLists.txt [hunk 1]
```diff
@@ -131,10 +132,26 @@ IF (${PACKAGE_NAME}_ENABLE_Epetra)
   IF(NOT ${${PACKAGE_NAME}_Epetra_NO_32BIT_GLOBAL_INDICES})
     MESSAGE("-- Xpetra support for 32 bit Epetra enabled.")
     MESSAGE("--    ${PACKAGE_NAME}_Epetra_NO_32BIT_GLOBAL_INDICES=" ${${PACKAGE_NAME}_Epetra_NO_32BIT_GLOBAL_INDICES})
+    
+    # Tpetra and Epetra need to match GO types
+    IF(${PACKAGE_NAME}_ENABLE_Tpetra)
+      IF(NOT Tpetra_INST_INT_INT)
+        MESSAGE(FATAL_ERROR "If Xpetra Epetra support is enabled and Epetra 32-bit global indices are enabled, Xpetra requres that you enable Tpetra_INST_INT_INT")
```
Comment (jhux2) [inline packages/xpetra/CMakeLists.txt]: "requres" -> "requires"

File: packages/xpetra/CMakeLists.txt [hunk 2]
```diff
@@ -131,10 +132,26 @@ IF (${PACKAGE_NAME}_ENABLE_Epetra)
...
+
   ENDIF()
   IF(NOT ${${PACKAGE_NAME}_Epetra_NO_64BIT_GLOBAL_INDICES})
     MESSAGE("-- Xpetra support for 64 bit Epetra enabled.")
     MESSAGE("--    ${PACKAGE_NAME}_Epetra_NO_64BIT_GLOBAL_INDICES=" ${${PACKAGE_NAME}_Epetra_NO_64BIT_GLOBAL_INDICES})
+
+    # Tpetra and Epetra need to match GO types
+    IF(${PACKAGE_NAME}_ENABLE_Tpetra)
+      IF(NOT Tpetra_INST_INT_LONG_LONG)
+        MESSAGE(FATAL_ERROR "If Xpetra Epetra support is enabled and Epetra 64-bit global indices are enabled, Xpetra requres that you enable Tpetra_INST_INT_LONG_LONG")
```
Comment (jhux2) [inline packages/xpetra/CMakeLists.txt]: "requres" -> "requires"

**Comment by csiefer2** [comment]
@trilinos/framework cuda test fails are due to machine configuration issue.  Retesting

**Comment by csiefer2** [comment]
Still having cuda issues

**Comment by mhoemmen** [comment]
b0rk'd: https://github.com/trilinos/Trilinos/issues/5147

**Pull request merged**